File: backend/src/main/java/com/project/entities/Assessment.java
package com.project.entities;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@Table(name = "assessments")
public class Assessment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long assessmentId;

    private String title;
    private String description;

    @OneToMany(mappedBy = "assessment", fetch = FetchType.LAZY)
    private List<AssessmentQuestion> questions;
}

File: backend_dotnet/SafeMind.Core/Entities/Assessment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SafeMind.Core.Entities
{
    [Table("assessments")]
    public class Assessment
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("assessment_id")]
        public long AssessmentId { get; set; }

        [Column("title")]
        public string? Title { get; set; }
        [Column("description")]
        public string? Description { get; set; }

        public ICollection<AssessmentQuestion> Questions { get; set; } = new List<AssessmentQuestion>();
    }
}
